hp.py

```diff
@@ -81,24 +78,19 @@ def show_dimm(self, raw_status):
     # Module #:                     4
     # Status:                       Ok
     #
-        dimm_status = {}
         dimm_list = []
+
         if raw_status is not None:
-            for line in raw_status.split('\n'):
-                new_line = line.split()
-                if new_line is not None:
-                    if "Processor" in new_line and "#:" in new_line:
-                        # new chunk - add the previous one to the list and create a new dict
-                        dimm_list.append(dimm_status)
```

Comment:
Doing it this way always means you have an empty dictionary as the first element in dimm_list.